src/services/codefixes/fixStrictClassInitialization.ts

```diff
@@ -0,0 +1,142 @@
+    const fixIdAddInitializer = "addMissingPropertyInitializer";
+    const errorCodes = [Diagnostics.Property_0_has_no_initializer_and_is_not_definitely_assigned_in_the_constructor.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions: (context) => {
+            const propertyDeclaration = getPropertyDeclaration(context.sourceFile, context.span.start);
+            if (!propertyDeclaration) return undefined;
+
+            const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+            const result = [
+                getActionsForAddMissingUndefinedType(context, propertyDeclaration),
+                getActionsForAddMissingDefiniteAssignmentAssertion(context, propertyDeclaration, newLineCharacter)
+            ];
+            let action: CodeFixAction | undefined;
+            if ((action = getActionsForAddMissingInitializer(context, propertyDeclaration, newLineCharacter))) result.push(action);
+            return result;
+        },
+        fixIds: [fixIdAddDefiniteAssignmentAssertions, fixIdAddUndefinedType, fixIdAddInitializer],
+        getAllCodeActions: context => {
+            const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+
+            return codeFixAll(context, errorCodes, (changes, diag) => {
+                const propertyDeclaration = getPropertyDeclaration(diag.file, diag.start);
+                if (!propertyDeclaration) return undefined;
+
+                switch (context.fixId) {
+                    case fixIdAddDefiniteAssignmentAssertions:
+                        addDefiniteAssignmentAssertion(changes, diag.file, propertyDeclaration, newLineCharacter);
+                        break;
+                    case fixIdAddUndefinedType:
+                        addUndefinedType(changes, diag.file, propertyDeclaration);
+                        break;
+                    case fixIdAddInitializer:
+                        const checker = context.program.getTypeChecker();
+                        const initializer = getInitializer(checker, propertyDeclaration);
+                        if (!initializer) return undefined;
+
+                        addInitializer(changes, diag.file, propertyDeclaration, initializer, newLineCharacter);
+                        break;
+                    default:
+                        Debug.fail(JSON.stringify(context.fixId));
+                }
+            });
+        },
+    });
+
+    function getPropertyDeclaration (sourceFile: SourceFile, pos: number): PropertyDeclaration | undefined {
+        const token = getTokenAtPosition(sourceFile, pos, /*includeJsDocComment*/ false);
+        return isIdentifier(token) ? cast(token.parent, isPropertyDeclaration) : undefined;
+    }
+
+    function getActionsForAddMissingDefiniteAssignmentAssertion (context: CodeFixContext, propertyDeclaration: PropertyDeclaration, newLineCharacter: string): CodeFixAction {

```

Comment:
Name the function `getActionFor` not `getActionsFor` if it doesn't return an array